block-producer/src/state_view/mod.rs

```diff
@@ -178,6 +178,8 @@ fn ensure_tx_inputs_constraints(
     debug!(target: COMPONENT, %tx_inputs);
 
     match tx_inputs.account_hash {
+        // Having `Some(store_account_hash)` here signifies that we are operating
+        // on the state of an existing account.
         Some(store_account_hash) => {
```

Comment:
I would probably make the comments more about "why" we are doing what we are doing. For example: "if the account is present in the store, make sure that the account state which went into the transaction is the same as the one in the store."